app/controllers/application_controller.rb

```diff
@@ -717,15 +717,25 @@ def build_not_found_page(status = 404, layout = false)
       layout = 'application' if layout == 'no_ember'
     end
 
-    category_topic_ids = Category.pluck(:topic_id).compact
-    @container_class = "wrap not-found-container"
-    @top_viewed = TopicQuery.new(nil, except_topic_ids: category_topic_ids).list_top_for("monthly").topics.first(10)
-    @recent = Topic.includes(:category).where.not(id: category_topic_ids).recent(10)
-    @slug =  params[:slug].class == String ? params[:slug] : ''
-    @slug =  (params[:id].class == String ? params[:id] : '') if @slug.blank?
+    key = "not_found_page_#{status}_#{layout}_#{!!current_user}"
+
+    if !(page = $redis.get(key))
```

Comment:
```suggestion
    unless page = $redis.get(key)
```

Reply:
I usually avoid `unless`, but in this case it looks better indeed.

Reply:
For the record, don't get me started on `unless` I always unconditionally prefer not to use it. I find `unless` a keyword I wish I could cancel from Ruby.